### ISSUE TITLE
New rule: fast-api-unused-path-parameter

### ISSUE THREAD

**Issue by Matthieu-LAURENT39** [astral-sh/ruff#12632]
Keywords: fastapi, path parameters

Hello, i've been using ruff for a while and i'm really loving it.
I saw that recently some FastAPI rules have been added, and i would like to suggest a new one, `fast-api-unused-path-parameter`.
It's pretty simple, it verifies that when defining a route that takes in path parameters, the function has matching arguments.

For example, the following code would trigger the rule, as we define a path parameter named "thing_id", but the function doesn't have an argument named "thing_id":
```py
from fastapi import FastAPI

app = FastAPI()

@app.get("/things/{thing_id}")
async def read_thing(id: int, query: str = None):
    return {"thing_id": id, "query": query}
```

Also of note is that it should ignore any characters after a `:` in the path parameter name, so this wouldn't trigger the rule:
```py
...

@app.get("/place/{my_path:path}")
def read_path(my_path):
    ...
```

If this rule is a good fit for ruff, i'd love to make a PR to implement it!

**Comment by killjoy1221** [comment]
How should routes with path parameters consumed in dependencies behave? Example:
```py
def foo(bar: Annotated[str, Path()]):
  print(bar)


@app.get("/foo/{bar}", dependencies=[Depends(foo)])
async def do_foo():
    pass
```

**Comment by MichaReiser** [comment]
Thanks for proposing a new rule.

Can you help me understand what's still missing from https://github.com/astral-sh/ruff/pull/12638 that shipped as preview rule in 0.6.1?
